File: Cpp/684_Redundant_Connection.cpp
class Solution {
public:
    vector<int> findRedundantConnection(vector<vector<int>>& edges) {
        vector<int> parents(1001);
        for(int i=0; i<parents.size(); ++i) {
            parents[i] = i;
        }
        
        for(auto edge : edges) {
            int v1 = find(parents, edge[0]);
            int v2 = find(parents, edge[1]);
            if (v1 == v2) return edge;
            parents[v1] = v2;
        }
        return {};
    }
    
    int find(vector<int>& parents, int x) {
        if(parents[x] != x) {
            parents[x] = find(parents, parents[x]);
        }
        return parents[x];
    }
};

File: Kotlin/684_Redundant_Connection.kt
class Solution {
    fun findRedundantConnection(edges: Array<IntArray>): IntArray {
        val parents = IntArray(1001)
        for(i in 0 until parents.size){
            parents[i] = i
        }
        
        for(edge in edges){
            val v1 = find(parents, edge[0])
            val v2 = find(parents, edge[1])
            if(v1 == v2) return edge
            parents[v1] = v2
        }
        
        return IntArray(0)
    }
    
    fun find(parents: IntArray, x: Int): Int {
        if(parents[x] != x) {
            parents[x] = find(parents, parents[x])
        }
        
        return parents[x]
    }
}